In [1]:
import pandas as pd

df = pd.read_csv("../data/raw/Bengaluru_House_Data.csv")

In [2]:
df.head()

,area_type,availability,location,size,society,total_sqft,bath,balcony,price
0,Super built-up Area,19-Dec,Electronic City Phase II,2 BHK,Coomee,1056,2.0,1.0,39.07
1,Plot Area,Ready To Move,Chikka Tirupathi,4 Bedroom,Theanmp,2600,5.0,3.0,120.00
2,Built-up Area,Ready To Move,Uttarahalli,3 BHK,NaN,1440,2.0,3.0,62.00
3,Super built-up Area,Ready To Move,Lingadheeranahalli,3 BHK,Soiewre,1521,3.0,1.0,95.00
4,Super built-up Area,Ready To Move,Kothanur,2 BHK,NaN,1200,2.0,1.0,51.00


In [3]:
df.shape

(13320, 9)

In [4]:
df.columns

Index(['area_type', 'availability', 'location', 'size', 'society',
       'total_sqft', 'bath', 'balcony', 'price'],
      dtype='str')

In [6]:

# Dataset Information

df.info()

<class 'pandas.DataFrame'>
RangeIndex: 13320 entries, 0 to 13319
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   area_type     13320 non-null  str    
 1   availability  13320 non-null  str    
 2   location      13319 non-null  str    
 3   size          13304 non-null  str    
 4   society       7818 non-null   str    
 5   total_sqft    13320 non-null  str    
 6   bath          13247 non-null  float64
 7   balcony       12711 non-null  float64
 8   price         13320 non-null  float64
dtypes: float64(3), str(6)
memory usage: 1.6 MB


In [7]:
df.describe()

,bath,balcony,price
count,13247.000000,12711.000000,13320.000000
mean,2.692610,1.584376,112.565627
std,1.341458,0.817263,148.971674
min,1.000000,0.000000,8.000000
25%,2.000000,1.000000,50.000000
50%,2.000000,2.000000,72.000000
75%,3.000000,2.000000,120.000000
max,40.000000,3.000000,3600.000000


In [8]:
df.describe(include="object")

C:\Users\Asus\AppData\Local\Temp\ipykernel_31768\702825166.py:1: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  df.describe(include="object")


,area_type,availability,location,size,society,total_sqft
count,13320,13320,13319,13304,7818,13320
unique,4,81,1305,31,2688,2117
top,Super built-up Area,Ready To Move,Whitefield,2 BHK,GrrvaGr,1200
freq,8790,10581,540,5199,80,843


In [9]:
df.isnull().sum()

area_type          0
availability       0
location           1
size              16
society         5502
total_sqft         0
bath              73
balcony          609
price              0
dtype: int64

In [10]:
df.nunique()

area_type          4
availability      81
location        1305
size              31
society         2688
total_sqft      2117
bath              19
balcony            4
price           1994
dtype: int64

In [11]:

# Missing Values

missing_values = pd.DataFrame({
    "Missing Values": df.isnull().sum(),
    "Percentage": round((df.isnull().sum()/len(df))*100,2)
})

missing_values.sort_values("Missing Values", ascending=False)

,Missing Values,Percentage
society,5502,41.31
balcony,609,4.57
bath,73,0.55
size,16,0.12
location,1,0.01
area_type,0,0.00
availability,0,0.00
total_sqft,0,0.00
price,0,0.00


In [12]:

# Data Types

pd.DataFrame(df.dtypes, columns=["Data Type"])

,Data Type
area_type,str
availability,str
location,str
size,str
society,str
total_sqft,str
bath,float64
balcony,float64
price,float64


In [13]:

# Memory Usage

df.memory_usage(deep=True)

Index              132
area_type       337506
availability    260771
location        278993
size            184858
society         162951
total_sqft      159910
bath            106560
balcony         106560
price           106560
dtype: int64

In [14]:

# Numerical Columns

numerical_columns = df.select_dtypes(include=["int64","float64"]).columns

print(numerical_columns)

Index(['bath', 'balcony', 'price'], dtype='str')


In [15]:

# Categorical Columns

categorical_columns = df.select_dtypes(include=["object"]).columns

print(categorical_columns)

Index(['area_type', 'availability', 'location', 'size', 'society',
       'total_sqft'],
      dtype='str')


C:\Users\Asus\AppData\Local\Temp\ipykernel_31768\2186242282.py:3: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  categorical_columns = df.select_dtypes(include=["object"]).columns


In [16]:

# Unique Values

for col in categorical_columns:
    print("="*60)
    print(col)
    print(df[col].nunique())

area_type
4
availability
81
location
1305
size
31
society
2688
total_sqft
2117


# Data Quality Summary

Total Rows :

Total Columns :

Missing Columns :

Duplicate Rows :

Numerical Columns :

Categorical Columns :

Potential Problems :

Cleaning Required :

Next Step :

In [18]:
# ==========================================================
# Project : AI House Price Prediction System
# Notebook: 01_dataset_understanding.ipynb
# Author  : Nishant Malik
# Purpose : Dataset Profiling and Initial Analysis
# ==========================================================

# Initial Dataset Analysis

## Dataset Information

- Dataset Name : Bengaluru House Price Dataset
- Total Records : 13,320
- Total Features : 9

## Target Variable

- price

## Numerical Features

- total_sqft
- bath
- balcony
- price

## Categorical Features

- area_type
- availability
- location
- size
- society

## Initial Problems Identified

- Missing values
- Duplicate values (to be verified)
- Mixed values in total_sqft
- Categorical columns require encoding
- High-cardinality location feature
- Society column may have many missing values

## Next Step

Perform complete Data Cleaning.